hail/src/main/scala/is/hail/annotations/UnsafeRow.scala

```diff
@@ -73,11 +73,11 @@ object UnsafeRow {
       case t: PArray =>
         readArray(t, region, offset)
       case t: PSet =>
-        readArray(t, region, offset).toSet
```

Comment:
I think these changes are wrong. What was the motivation?

In particular, if you have a `PSet[PLocus]`, this is going to return a `Set[Row]` (the struct representation of locus)`